src/actions.ml

```diff
@@ -628,76 +672,536 @@ let minimize_failed_tests ~bot_info ~owner ~repo ~pr_number ~base ~head
                       if String.is_prefix name ~prefix:"test-suite" then
                         Some name
                       else None)
+              in
+              let possible_jobs_to_minimize, unminimizable_jobs =
+                head_checks
+                |> List.partition_map ~f:(fun (({name}, _) as head_check) ->
+                       match
+                         ci_minimization_fetch_suggestion_info
+                           ~head_pipeline_summary ~base_pipeline_summary
+                           ~base_checks_errors ~base_checks head_check
+                       with
+                       | Error err ->
+                           Either.Second (name, err)
+                       | Ok result ->
+                           Either.First result)
+              in
+              Lwt.return_ok
+                ( { comment_thread_id= pr_id
+                  ; base
+                  ; head
+                  ; pr_number
+                  ; draft= false (* XXX FIXME *)
+                  ; labels= [] (* XXX FIXME *)
```

Comment:
You need this patch to the GraphQL request and to adapt the `GitHub_queries` module correspondingly:

```diff
diff --git a/bot-components/GitHub_GraphQL.ml b/bot-components/GitHub_GraphQL.ml
index 90ff7a3..040159d 100644
--- a/bot-components/GitHub_GraphQL.ml
+++ b/bot-components/GitHub_GraphQL.ml
@@ -357,6 +357,12 @@ query getChecks($appId: Int!, $owner: String!, $repo:String!, $prNumber: Int!, $
   repository(name: $repo,owner:$owner) {
     pullRequest(number: $prNumber) {
       id
+      labels(first: 100) {
+        nodes {
+          name
+        }
+      }
+      isDraft
     }
     base: object(expression: $base) {
       ... on Commit {
```

Reply:
Thanks!